ipaplatform/redhat/authconfig.py

```diff
@@ -18,15 +18,146 @@
 # You should have received a copy of the GNU General Public License
 # along with this program.  If not, see <http://www.gnu.org/licenses/>.
 
+import logging
+import six
+import abc
+import re
+
 from ipaplatform.paths import paths
 from ipapython import ipautil
 from ipapython.admintool import ScriptError
 import os
 
 FILES_TO_NOT_BACKUP = ['passwd', 'group', 'shadow', 'gshadow']
 
+logger = logging.getLogger(__name__)
+
+
+@six.add_metaclass(abc.ABCMeta)
+class RedHatAuthTool(object):
+
+    @staticmethod
+    def get_tool():
+        return RedHatAuthSelect()
+
+    @abc.abstractmethod
+    def configure(self, sssd, mkhomedir, statestore):
+        pass
+
+    @abc.abstractmethod
+    def unconfigure(self, fstore, statestore,
+                    was_sssd_installed,
+                    was_sssd_configured):
+        pass
+
+    @abc.abstractmethod
+    def backup(self, path):
+        pass
+
+    @abc.abstractmethod
+    def restore(self, path):
+        pass
+
+
+class RedHatAuthSelect(RedHatAuthTool):
+
+    def __get_authselect_current_output(self):
+        current = ipautil.run(
+            [
+                paths.AUTHSELECT,
+                "current"
+            ],
+            env={"LC_ALL": "C.UTF8"}
+        )
+
+        if current.returncode != 0:
+            return None
+
+        return current.raw_output.decode()
+
+    def __parse_authselect_output(self, output_text=None):
+        if output_text is None:
+            output_text = self.__get_authselect_current_output()
+
+        if output_text is None:
+            return None
+
+        cfg_params = re.findall(
+            r"\s*Profile ID:\s*(\S+)\s*\n\s*Enabled features:\s*(.*)",
+            output_text,
+            re.DOTALL
+        )
+
+        profile = cfg_params[0][0]
+
+        if not profile:
+            return None
+
+        features = re.findall("-\s*(\S+)", cfg_params[0][1], re.DOTALL)
+
+        return (profile, features)
 
-class RedHatAuthConfig(object):
+    def configure(self, sssd, mkhomedir, statestore):
+        if not sssd:
+            logger.error("no-sssd options is not supported by installer. "
+                         "For client installation without sssd check the "
+                         "output of 'ipa-advise config-fedora-authconfig'")
+            return
+            # raise ScriptError("Non-sssd installation is not supported")
+
+        cfg = self.__parse_authselect_output()
+        if cfg:
+            statestore.backup_state('authselect', 'profile', cfg[0])
+            statestore.backup_state('authselect', 'features_list',
+                                    " ".join(cfg[1]))
```

Comment:
It's easier readable if you move all three arguments into the same line.

Reply:
What is easier to read is a very subjective thing. Anyway, the line then will not fit into 80 chars.

Reply:
All arguments fit easily on one line:
```
        statestore.backup_state(
            'authselect', 'features_list', " ".join(cfg[1]))
```

In case you have to spread a function call over more than one line, use hanging indention and try to minimize the indention level of the function arguments.